{{cookiecutter.project_slug}}/config/settings/production.py

```diff
@@ -87,22 +87,22 @@
 STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'
 {%- else %}
 STATICFILES_STORAGE = 'config.settings.production.StaticRootS3BotoStorage'
-STATIC_URL = f'https://s3.amazonaws.com/{AWS_STORAGE_BUCKET_NAME}/static/'
+STATIC_URL = f'https://{AWS_STORAGE_BUCKET_NAME}.s3.amazonaws.com/static/'
```

Comment:
S3 buckets are reachable via subdomain (like you've got here) and `/` path notation alike. Therefore, the variant you propose is more generic which is a definitive plus.